### PR TITLE
A new branch for Scala 2.13

### PR DESCRIPTION
**Pull request by aluscent**
This pull-request will address #8

### CODE REVIEW

**Comment by jtnystrom** [comment]
Hi Ali, thanks a lot for this! This looks good, except that the commit message "update scala version to 2.12.18" should probably be 2.13.12, right? Do you think you could fix that (I'm not sure about the easiest way, either in this PR or by pushing a new one if you have to?)

**Comment by aluscent** [comment]
Thanks for mentioning this. I should've been mistaken.
I made some changes. Could you please see if it's ok?

File: src/main/scala/bio/alignment/alignment.scala [hunk 1]
```diff
@@ -85,10 +88,10 @@ trait SparseAlignment[T] {
   /**
    * Remove columns, i.e. columns that contain less than min_symbols
    */
-  def removeSparseColumns(in: Alignment, min_symbols: Int): (Alignment, List[Int]) = {
+  def removeSparseColumns(in: List[List[T]], min_symbols: Int): (Alignment, List[Int]) = {
```
Comment (jtnystrom) [inline src/main/scala/bio/alignment/alignment.scala:91]: Why change the type from Alignment to List[List[T]]? I think the former is more readable given how the rest of the API for SparseAlignment is designed.

File: src/main/scala/bio/sequence/codonsequence.scala [hunk 2]
```diff
@@ -65,17 +65,15 @@ package Protein {
         seq.grouped(3).toSeq
 
       // Amino acids and nucleotides
-      val aas = ToSequence(str).translate() // IUPAC Sequence
```
Comment (jtnystrom) [inline src/main/scala/bio/sequence/codonsequence.scala:68]: Why remove this comment? (And similar below)
Does it add no new information in the context?

**Comment by jtnystrom** [comment]
Thanks a lot. Looks like a lot of good style changes overall, and the branch compiles for me and tests are passing.  I added two minor comments about things I didn't understand.

File: src/main/scala/bio/alignment/alignment.scala (hunk 1 above)
Comment (aluscent) [reply src/main/scala/bio/alignment/alignment.scala:91]: It should've been for test porposes. I corrected it.

File: src/main/scala/bio/sequence/codonsequence.scala (hunk 2 above)
Comment (aluscent) [reply src/main/scala/bio/sequence/codonsequence.scala:68]: I believe in code being self-explanatory, that's why I am against commenting. 
In this case, I re-checked the code now and I might mistaken them for commented code.
I will roll-back these changes.

**Comment by jtnystrom** [comment]
Looks good, thanks!

File: src/main/scala/bio/sequence/codonsequence.scala (hunk 2 above)
Comment (jtnystrom) [reply src/main/scala/bio/sequence/codonsequence.scala]: Yep, I agree with you. In this case I think it was ambiguous, so either way would probably have been fine in the end.

**Pull request merged**